PyInstaller/hooks/hook-PySide2.QtWebEngineWidgets.py

```diff
@@ -1,51 +1,101 @@
-             os.path.join('QtWebEngineProcess.app', 'Contents', 'MacOS'))
-        ]
+def get_relative_path_if_possible(actual, possible_prefix):
+    actual_path = os.path.abspath(actual)
+    possible_prefix_path = os.path.abspath(possible_prefix)
+    possible_relative_path = os.path.relpath(actual_path, possible_prefix_path)
+    if possible_relative_path.startswith(os.pardir):
+        return actual
+    else:
+        return possible_relative_path
+
+
+# Ensure PySide2 is importable before adding info depending on it.
+if pyside2_library_info.version:
+    hiddenimports, binaries, datas = add_qt5_dependencies(__file__)
 
-        resources_dir = os.path.join(libdir, 'QtWebEngineCore.framework', 'Versions', '5', 'Resources')
+    # Include the web engine process, translations, and resources.
+    # According to https://bugreports.qt.io/browse/PYSIDE-642
+    # there's no subdir for windows
+    if compat.is_win:
+        rel_data_path = ['PySide2']
+    else:
+        rel_data_path = ['PySide2', 'Qt']
+
+    pyside2_locations = pyside2_library_info.location
+    if compat.is_darwin:
+        # This is based on the layout of the Mac wheel from PyPi.
+        data_path = pyside2_locations['DataPath']
+        libraries = ['QtCore', 'QtWebEngineCore', 'QtQuick', 'QtQml',
+                     'QtNetwork', 'QtGui', 'QtWebChannel',
+                     'QtPositioning']
+        for i in libraries:
+            datas += collect_system_data_files(
+                os.path.join(data_path, 'lib', i + '.framework'),
+                os.path.join(*(rel_data_path + ['lib'])), True)
+        datas += [(os.path.join(data_path, 'lib', 'QtWebEngineCore.framework',
+                                'Resources'), os.curdir)]
+    else:
+        locales = 'qtwebengine_locales'
+        resources = 'resources'
         datas += [
-            (os.path.join(resources_dir, 'icudtl.dat'), '.'),
-            (os.path.join(resources_dir, 'qtwebengine_resources.pak'), '.'),
-            # The distributed Info.plist has LSUIElement set to true, which prevents the
-            # icon from appearing in the dock.
-            (os.path.join(libdir, 'QtWebEngineCore.framework', 'Versions', '5',
-                          'Helpers', 'QtWebEngineProcess.app', 'Contents', 'Info.plist'),
-             os.path.join('QtWebEngineProcess.app', 'Contents'))
+            # Gather translations needed by Chromium.
+            (os.path.join(pyside2_locations['TranslationsPath'],
+                          locales),
+             os.path.join(*(rel_data_path + ['translations',
+                                             locales]))),
+            # Per the `docs
+            # <https://doc.qt.io/qt-5.10/qtwebengine-deploying.html#deploying-resources>`_,
+            # ``DataPath`` is the base directory for ``resources``.
+            #
+            # When Python 3.4 goes EOL (see `PEP 448`_, this is better written
+            # as ``os.path.join(*rel_data_path, resources)``.
```

Comment:
You could put the old code into a `if not is py_2` block (and the new code into the else-part), this it is easy to spot and eliminate the old code later. (Py 3.4 is EOL already and support has already been remved from PyInstaller, thus we only need to take care about Py 2 here.)

(This of course only works if Py 2.7 accepts teh syntax.)

Reply:
Moved the path join calls pertaining to the new syntax to a separate function.